lib/wallaroo/core/source/tcp_source/tcp_source.pony

```diff
@@ -126,8 +127,10 @@ actor TCPSource is Producer
     _route_builder = route_builder
     for (target_worker_name, builder) in outgoing_boundary_builders.pairs() do
       if not _outgoing_boundaries.contains(target_worker_name) then
-        _outgoing_boundaries(target_worker_name) =
+        let new_boundary =
           builder.build_and_initialize(_step_id_gen(), _layout_initializer)
+        router_registry.register_disposable(new_boundary)
+        _outgoing_boundaries(target_worker_name) = new_boundary
```

Comment:
can you open an issue for moving this responsibility down a level to generic source?

Reply:
I already created an issue about cleaning up our approach to disposables.  I'm going to add it onto that one